rfcs/12-primitive-shapes.md

```diff
@@ -0,0 +1,545 @@
+
+```rust
+pub struct Circle2d {
+  radius: f32,
+}
+```
+
+Note that a `Circle2d` does not contain any information about the position of the circle. This is due to how shapes are composed to add more complex functionality. Consider the following common use cases of primitives shapes, as well as the the information (Translation and Rotation) that are needed to fully define the shapes for these cases:
+
+| Shape     | Mesh | Bounding | Collision |
+|---        |---|---|---|
+| Sphere    | ✔ | ✔  + Trans | ✔ + Trans |
+| Box    | ✔ | ✔ + Trans (AABB) | ✔ + Trans + Rot (OBB) |
+| Capsule   | ✔ | ❌ | ✔ + Trans + Rot |
+| Cylinder  | ✔ | ❌ | ✔ + Trans + Rot |
+| Cone      | ✔ | ❌ | ✔ + Trans + Rot |
+| Wedge     | ✔ | ❌ | ✔ + Trans + Rot |
+| Plane     | ✔ | ❌ | ✔ |
+| Torus     | ✔ | ❌ | ❌ |
+
+### Bounding vs. Collision
+
+The difference between the two is somewhat semantic. Both bounding and collision check for intersections between bounding volumes or areas. However, bounding volumes are generally used for broad phase checks due to their speed and small size, and are consequently often used in tree structures for spacial queries. Note that both bounding types are not oriented - this reduces the number of comparisons needed to check for intersections during broad phase tests. Bounding spheres or AABBs are generally always preferred for broad phase bounding checks, despite the OBB's name.
+
+**Note that the Bounding/Collision interface is not part of this RFC**, the purpose of this discussion is to present common use cases of these primitive shape types to uncover deficiencies in their design.
+
+### Where are the `Transform`s?
+
+In this potential bounding and collision interface, translation and rotation are **not** defined using bevy's `Transform` components, nor are they stored in the base primitive shape types. This is because types that implement `Bounding` and `Collider` must be fully self-contained. This:
+ 
+ * makes the API simpler when using these components in functions and systems
+ * ensures bounding and collision types use an absolute minimum of memory
+ * prevents errors caused by nonuniform scale invalidating the shape of the primitive.
+
+Instead, changes to the parent's `GlobalTransform` should be used to derive a new primitive on scale changes - as well as a new translation and rotation if required.
+
+Although bounding and collision are out of scope of this RFC, this helps to explain why primitive shape types contain no positional information. This hypothetical bounding/collision interface shows how the base primitive shapes can be composed to build common game engine functinality without adding memory overhead.
+
+### Meshing
+
+Both 2d and 3d primitives could implement the `Meshable` trait to provide the ability to generate a tri mesh:
```

Comment:
This is something I considered earlier on, but there were some questions about what kind of mesh and what density of mesh is generated when you call that. We could do both though! `From<Shape>` would be super handy for simple immediate-mode style debugging where you just want a low poly representation, and a trait like `Meshable` could be handy to provide a standard interface for generating meshes with a specified density. I definitely agree that using from/into would be a really nice rusty solution for some use cases.

Reply:
It's helpful to have a parameter to control the detail of the mesh generated, since it's an approximation of the limit surface. This allows for different levels of detail, which can be adjusted based on size or distance from the camera.